casepropods/family_connect_registration/plugin.py

```diff
@@ -37,56 +74,177 @@ def lookup_field_from_dictionaries(self, field, *lists):
-
+    def get_identity_registration_data(self, identity, registrations):
+        """
+        Given an identity and a list of registrations, formats the registration
+        data into the pod format.
+        """
         if identity is not None and 'details' in identity:
             identity_details = [identity['details']]
         else:
             identity_details = []
 
-        for field in mapping:
+        registration_data_fields = [r['data'] for r in registrations]
+
+        items = []
+        for field in self.config.field_mapping:
             value = self.lookup_field_from_dictionaries(
                 field['field'],
                 identity_details,
-                results,
-                results_data_field,
+                registrations,
+                registration_data_fields,
             )
-            content['items'].append({
+            items.append({
                 'name': field['field_name'],
                 'value': value,
             })
+        return items
+
+    def has_whatsapp_account(self, number):
+        """
+        Checks if the given number has a registered whatsapp account, using the
+        wassup API
+        """
+        res = requests.get(
+            self.config.wassup_url,
+            {
+                'number': self.config.wassup_number,
+                'address': number,
+                'wait': True,
+            },
+            headers={
+                'Authorization': 'Token {}'.format(self.config.wassup_token),
+            },
+        )
+        res.raise_for_status()
+        res = res.json()
+        existing = filter(lambda d: d.get('exists', False), res.values())
+        return any(existing)
+
+    def get_switch_channel_action(self, channel, identity):
+        """
+        Given the current channel that the user is on, returns a pod action
+        for switching to the other channel choice
+        """
+        opposite_channel = 'whatsapp' if channel == 'sms' else 'sms'
+        opposite_label = 'WhatsApp' if channel == 'sms' else 'SMS'
+        return {
+            'type': 'switch_channel',
+            'name': 'Switch to {}'.format(opposite_label),
+            'confirm': True,
+            'busy_test': 'Processing...',
+            'payload': {
+                'channel': opposite_channel,
+                'channel_label': opposite_label,
+                'identity': identity['id'],
+            },
+        }
+
+    def get_address_from_identity(self, identity):
+        """
+        Given an identity from the identity store, returns the active address
+        for that identity. If no address exists, returns None
+        """
+        addresses = identity.get(
+            'details', {}).get('addresses', {}).get('msisdn', {})
+        msisdn = None
+        for address, data in addresses.items():
+            if data.get('optedout', False):
+                continue
+            if data.get('default', False):
```

Comment:
If default is True, you continue to next loop without setting 'msisdn'. Maybe add a test where the identity has a default msisdn.

Reply:
Ah yes, good find. I think this is supposed to just `return address` if `default` is `True`. I'll add a fix and a test